### PR TITLE
[WIP] Kahan Summation in Translator

### PR DESCRIPTION
**Pull request by mio-stripe**
Adding Kahan summation of Exprs in Translator.

### CODE REVIEW

**Comment by mio-stripe** [comment]
So far, the SBC's fail as expected, but produce reasonable numbers.
I get GC overloads with even small numbers of observations and traces show this is generating _tons_ of code so plenty of work to do.

**Comment by avibryant** [comment]
I pushed https://github.com/stripe/rainier/tree/avi/ks-in-ir which merges in the current develop (to get the graphviz stuff) and makes a change that I think fixes the excess code gen without screwing up correctness, but I haven't done any real testing so please verify.

File: rainier-core/src/main/scala/com/stripe/rainier/compute/Translator.scala [hunk 1]
```diff
@@ -171,11 +171,27 @@ private class Translator {
     combineTree(lazyExprs, ring)
   }
 
+  private def kahanSumUpdate(accumulator: Expr,
+                             compensation: Expr,
+                             term: Expr) = {
+    val nextExpr = binaryExpr(term, compensation, SubtractOp)
+    val newAccumulator = binaryExpr(accumulator, nextExpr, AddOp)
```
Comment (mio-stripe) [inline rainier-core/src/main/scala/com/stripe/rainier/compute/Translator.scala]: With `ref(accumulator)` here, I get an uninitialized register error. As is, even with these other `refs` I still get hanging GC.

Comment (avibryant) [reply rainier-core/src/main/scala/com/stripe/rainier/compute/Translator.scala]: Have you done graphviz plots to visualize what's going on?

**Comment by mio-stripe** [comment]
So it turns out the GC overhead exceeded is because Packer is doing some way-too-deep recursion so this is not the way to go. Closing.

**Pull request closed without merging**